src/lib/utils/synapseTypes/Table/QueryFilter.ts

```diff
@@ -1,29 +1,67 @@
 }
 export enum ColumnMultiValueFunction {
   HAS = 'HAS', // HAS function on multi-value columns, same as the predicate: 'columnName HAS ()'
-  HAS_LIKE = 'HAS_LIKE' // HAS_LIKE function on multi-value columns, same as the predicate: 'columnName HAS_LIKE ()'. Can be used for pattern matching using wildcard %.
+  HAS_LIKE = 'HAS_LIKE', // HAS_LIKE function on multi-value columns, same as the predicate: 'columnName HAS_LIKE ()'. Can be used for pattern matching using wildcard %.
 }
 
+const COLUMN_SINGLE_VALUE_QUERY_FILTER_CONCRETE_TYPE_VALUE =
+  'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter'
+type COLUMN_SINGLE_VALUE_QUERY_FILTER_CONCRETE_TYPE =
+  typeof COLUMN_SINGLE_VALUE_QUERY_FILTER_CONCRETE_TYPE_VALUE
+
 // A filter using a single value against a column. Values within the values list have an OR relationship
 export interface ColumnSingleValueQueryFilter extends QueryFilter {
-  concreteType: 'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter'
+  concreteType: COLUMN_SINGLE_VALUE_QUERY_FILTER_CONCRETE_TYPE
+  columnName: string // name of the column to filter
   operator: ColumnSingleValueFilterOperator // Determines the filter operation to perform
   values: string[] // Values to used with the filter.
 }
 
+const COLUMN_MULTI_VALUE_FUNCTION_QUERY_FILTER_CONCRETE_TYPE_VALUE =
+  'org.sagebionetworks.repo.model.table.ColumnMultiValueFunctionQueryFilter'
+type COLUMN_MULTI_VALUE_FUNCTION_QUERY_FILTER_CONCRETE_TYPE =
+  typeof COLUMN_MULTI_VALUE_FUNCTION_QUERY_FILTER_CONCRETE_TYPE_VALUE
+
 // A filter that applies multi-value columns functions.
 export interface ColumnMultiValueFunctionQueryFilter extends QueryFilter {
-  concreteType: 'org.sagebionetworks.repo.model.table.ColumnMultiValueFunctionQueryFilter'
+  concreteType: COLUMN_MULTI_VALUE_FUNCTION_QUERY_FILTER_CONCRETE_TYPE
+  columnName: string // name of the column to filter
   function: ColumnMultiValueFunction // Determines the filter operation to perform
   values: string[] // Values to used with the filter.
 }
 
+// https://docs.synapse.org/rest/org/sagebionetworks/repo/model/table/TextMatchesQueryFilter.html
+const TEXT_MATCHES_QUERY_FILTER_CONCRETE_TYPE_VALUE =
+  'org.sagebionetworks.repo.model.table.TextMatchesQueryFilter'
+type TEXT_MATCHES_QUERY_FILTER_CONCRETE_TYPE =
+  typeof TEXT_MATCHES_QUERY_FILTER_CONCRETE_TYPE_VALUE
+export interface TextMatchesQueryFilter extends QueryFilter {
+  concreteType: TEXT_MATCHES_QUERY_FILTER_CONCRETE_TYPE
+  searchExpression: string
+}
+
 export interface QueryFilter {
-  concreteType: string
-  columnName: string // name of the column to filter
-  operator?: ColumnSingleValueFilterOperator
-  function?: ColumnMultiValueFunction
-  values: string[]
+  concreteType:
+    | TEXT_MATCHES_QUERY_FILTER_CONCRETE_TYPE
+    | COLUMN_SINGLE_VALUE_QUERY_FILTER_CONCRETE_TYPE
+    | COLUMN_MULTI_VALUE_FUNCTION_QUERY_FILTER_CONCRETE_TYPE
 }
```

Comment:
The majority of the changes in this PR are because it's no longer true that a QueryFilter has a `values` field. Thankfully, we have static typing!